src/main/java/bowling/domain/frame/Frame.java

```diff
@@ -0,0 +1,64 @@
+package bowling.domain.frame;
+
+import bowling.domain.Round;
+import bowling.domain.Pin;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class Frame {
+    public static final int FIRST_PLAY = 1;
+    public static final int SECOND_PLAY = 2;
+    public static final int BONUS_PLAY = 3;
+    public static final int FIRST_PIN = 0;
+    public static final int SECOND_PIN = 1;
+    public static final int BONUS_PIN = 2;
+
+    protected List<Pin> pins;
+
+    public Frame() {
+        pins = new ArrayList<>();
+    }
+
+    abstract public Frame next(Round round);
+
+    abstract public void play(int countOfDownPin);
+
+    public abstract boolean isEnd();
+
+    public abstract Boolean isNextFrame();
+
+    public int getPinsSize() {
+        return pins.size();
+    }
+
+    public int getTotalPinCount() {
+        return pins.stream()
+                .map(Pin::getPin)
+                .reduce(Pin.MIN_PIN_COUNT, Integer::sum);
```

Comment:
이렇게도 표현해 볼 수 있을 것 같습니다 😄 
이 부분도 참고만 부탁 드리겠습니다 🙇 
```suggestion
        return pins.stream()
                .mapToInt(Pin::getPin)
                .sum();
```